### DISCUSSION TITLE
Scaling a table to fit

### DISCUSSION THREAD

**Discussion by tboby** [QuestPDF/QuestPDF#73] [Q&A] [answered]
I'm trying to build an internally shared project using this library in a context where I don't want paging. It's going really well on the whole; `ShowEntire()` catches most layout problems and the size param on `Canvas()` makes it easy to generate embedded content at the right size.

The problem I'm running into is with `Stack`s where I don't know for sure the exact number of items that will there at runtime. I can pick font sizes and paddings that work for the expected number of rows, but if a user adds an extra category then they will get a `DocumentLayoutException`. I can compensate by designing for a greater number of rows, but then that wastes space.

I've been trying to think of ways of handling this and what comes to mind:
1. `Scale`. Simply scaling down seems to handle this very well. The problem is that I can't find a way to scale at runtime, and `DocumentLayoutException` isn't thrown in the context where I set the scale param so I can't catch and try a smaller scale.
2. A fallback layout. If `ShowEntire` fails, then try again with a different layout which is more bespoke (something closer to responsive breakpoints that allow choosing what to make smaller). But again, I'm not sure I can detect when to do this.
3. Using `Canvas` and just designing the table to the size given using SkiaSharp directly, but having to reproduce all the handy features like font styles.

I know it would be much better if I could either allow paging or have a fixed limit on collections, but I'd be nice if there was a safe fallback.

Any suggestions on ways I might be able to do this?

**Comment by MarcinZiabek** [comment]
Your use-case will be easily solvable in one of the future releases with the `DynamicComponent` functionality. But it still is in the experimental phase. You can read more about the current design [in this discussion](https://github.com/QuestPDF/QuestPDF/discussions/39).

Right now, I can suggest finding any heuristics. E.g. simple method, that based on the number of items, returns an approximation of proper Scale value to use or target text size - in such a way that the content will fit on the page. It will not be perfect but should work.

**Comment by DawidRobertNowak** [comment, marked as answer]
Hello,

We are closing this thread due to inactivity. Thank you for taking the time to reach out and contribute to the project — your feedback helps us improve QuestPDF.

Since this discussion was opened, QuestPDF has introduced both the `ScaleToFit` element (which automatically scales content down to fit available space) and [Dynamic Components](https://www.questpdf.com/concepts/code-patterns/dynamic-components.html) (which allow runtime content decisions based on available space). Either approach could address the use case you described.

Should you encounter any further problems, please don't hesitate to open a new issue.

Thank you!
